020-Getting-started/110-nuxt.mdx

```diff
@@ -0,0 +1,597 @@
+
+![Posts table](images/xata-table-posts.png)
+
+Click **Schema** to see the schema definition with the inferred data types:
+
+![Posts schema](images/xata-schema-posts.png)
+
+You'll also see `xata.*` [special columns](https://xata.io/docs/concepts/data-model#special-columns) automatically
+created and maintained by Xata.
+
+With the database schema in place, the final step is to generate the code that allows us to access and query the data
+from our Nuxt application. To do this, run:
+
+```sh
+xata pull main
+```
+
+<Expand title="xata pull main output">
+
+```sh
+$ xata pull main
+Successfully pulled 1 migrations from main branch
+Running codegen...
+Generated Xata code to ./src/xata.ts
+```
+
+</Expand>
+
+This updates the contents of `src/xata.ts` based on the schema defined on the `main` branch of your database. So, if you
+make any further changes to the schema, run `xata pull <branch>` to update the auto-generated code.
+
+## Basic styling and layout
+
+Begin by following step 2 onwards of the [Tailwind CSS and Nuxt 3 integration guide](https://tailwindcss.com/docs/guides/nuxtjs#3).
+Once completed, add the following global styling to the newly created `assets/css/main.css`:
+
+```css title="assets/css/main.css" {5-26}
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
+
+:root {
+  --foreground-rgb: 0, 0, 0;
+  --background-start-rgb: 214, 219, 220;
+  --background-end-rgb: 255, 255, 255;
+}
+
+@media (prefers-color-scheme: dark) {
+  :root {
+    --foreground-rgb: 255, 255, 255;
+    --background-start-rgb: 0, 0, 0;
+    --background-end-rgb: 0, 0, 0;
+  }
+}
+
+body {
+  color: rgb(var(--foreground-rgb));
+  background: linear-gradient(to bottom, transparent, rgb(var(--background-end-rgb))) rgb(var(--background-start-rgb));
+}
+```
+
+Finally, update `app.vue` to add some shared structure across application pages. The code will end up as follows:
+
+```tsx title="app.vue"
+<script setup lang="ts">
+useHead({
+  title: 'Get started with Xata and Nuxt',
+})
+</script>
+<template>
+  <main class="flex flex-wrap justify-center content-start p-8 lg:p-24 min-h-screen">
+    <div class="z-10 h-50 w-full max-w-5xl items-center justify-between text-xl lg:flex">
+      <p class="fixed left-0 top-0 flex w-full justify-center pb-6 pt-8 lg:static lg:w-auto bg-gradient-to-b from-white via-white via-65% dark:from-black dark:via-black lg:bg-none">
+        <a href="/">Get started with Xata and Nuxt</a>
+      </p>
+      <div class="fixed bottom-0 left-0 flex w-full items-end justify-center bg-gradient-to-t from-white via-white dark:from-black dark:via-black lg:static lg:h-auto lg:w-auto lg:bg-none">
+        <a href="https://xata.io" class="w-20">
+          <img src="https://raw.githubusercontent.com/xataio/examples/main/docs/app_logo.svg" />
+        </a>
+      </div>
+    </div>
+    <NuxtPage />
+  </main>
+</template>
+```
+
+Ensure the `<NuxtPage />` component is before the closing `</main>` element.
```

Comment:
I've never used Nuxt before. When I initially started up using `npm run dev`, it started at port `3003` (I had other stuff running, no bigs), but wouldn't register changes to this page. My guess is that this was because of the order of operations. Running `npm run dev` before doing the tailwind changes / config changes caused Nuxt not to pick up the changes.

Easy fix is to not suggest running `npm run dev` until after tailwind is configured (suggestion). Optionally, remind people to restart their server at this point.

Reply:
Added note.